### DISCUSSION TITLE
Blog posts and tutorials for newcomers

### DISCUSSION THREAD

**Discussion by ykafia** [stride3d/stride#1211] [Contributors]
So i made [a little plan](https://github.com/ykafia/gamedev-blogs/blob/master/plan.md) for a series of 4 blog posts to introduce newcomers to Stride and gamedev.
Any thoughts about it ?

**Comment by VaclavElias** [comment]
I like your content! 

- Introduction - Cool
- Stride Level 1: A simple game - This could be literally A-Z, even it is a Level 1, I would suggest to include
  - Showing how to use GitHub Repo for this game (but not teaching git, GitHub)
  - Deployment at the end
  - Publishing the game to Steam (just overview and basics)
  - Also, you need to decide your audience?
     1. never used C# and git
     2. good C# and git knowledge
  - The youtube/twitch videos would be cool to see the whole development, maybe to save your time, you don't need to explain everything code wise based on the audience level you target?
- Stride Level 2: SDSL - This would be for me, I have no clue on any on this subject
- Stride Level 3: The Unity tutorials is going to be cool

Other suggestions for future tutorials:
- This could be another whole series - Using open source Blender, creating basic shapes and whatever and porting, implementing in Stride

I started working personally on some random tutorials https://github.com/VaclavElias/Stride3DTutorials, well at the moment I have got only code but would like to make it as tutorial. 

The one which really matters to me is Minimal API, I would like to create some template to have Minimal Stride setup like in .NET https://docs.microsoft.com/en-us/aspnet/core/tutorials/min-web-api?view=aspnetcore-6.0&tabs=visual-studio, including suggesting some PR for Stride to simplify this for certain audience.

This could lead to very cool C# Beginner Tutorials, where instead of using boring console, you would use Game Engine as visual output, e.g. one doesn't need to know anything about games, but as a C# Beginner you could have e.g. output some 3D boxes in the engine, and create loops to display more boxes, including using basic math to position boxes and so on..

**Reply by ykafia** [reply]
@VaclavElias 
You're very right, I should add basic git, but i think for C# it's better to use [tutorials Microsoft made](https://dotnet.microsoft.com/learn/dotnet/in-browser-tutorial/1)

I love the idea of art pipeline tutorial, i think it's very important. Surely for a next series!

When you say minimal api you mean sandbox games ?

(Sorry for spamming, i guess answers are better than other comments)

**Reply by VaclavElias** [reply]
Yes, I agree, Microsoft has very good tutorials now!

I tried to use Blender and it really looks cool but I didn't know how to properly, easily move the assets to Unity, so it would be a nice thing to have for Stride, again not teaching Blender but showing something like, let's create this game asset(s) in Blender and then move them to Stride, and if it was a Video Tutorial, then it would be helpful saying what shortcuts you are using on Keyboard for each step, or use some application which shows it on the screen...

By minimal API, I mean a minimum setup for Stride to start using it. I am going to paste a screen shot of a code example which doesn't work. I have got actually also some working code from our Discord, where we call it code only which I just need to put together. So a C# Tutorial for Beginners wouldn't use a Console with "Hello, World" example output but Stride which would be more cool, because on the way of learning C# you could learn basics of Stride and game development, or just 2D or 3D basics.

![image](https://user-images.githubusercontent.com/4528464/144722796-68daa56e-3dd4-45c9-8db4-3aa7cd1fbbf6.png)

**Comment by ykafia** [comment]
So i've managed to write up an introduction, would love some opinions on what to make better, add or remove

https://github.com/ykafia/gamedev-blogs/blob/master/Introduction.md


I'll work on the first full game tutorial, might be both a pong and a tic-tac-toe depending the amount of work. But i'm going to do a short one about how to use blender with Stride since i'm going to create assets from blender

**Reply by VaclavElias** [reply]
I like that you don't go to too much details in the Introduction because anybody can just Google it and you put it together nicely and that is for me personally enough. Nicely and simple explained what is what and how it all fits together. 

I do have strong C# experience but regarding games, game jargon & steps I am lost so everything is over the place and so any such tutorial is going to be very helpful for me.  

Thanks!

**Comment by Aggror** [comment]
Hey, how is it going with this project?

**Reply by ykafia** [reply]
It's advancing!
I'm almost finished with a pong game, but I want to make it beautiful with nice assets and such!
